taui/src/components/anonymous-authenticator.js

```diff
@@ -0,0 +1,72 @@
+// @flow
+import { Component, Fragment } from 'react'
+import LogRocket from 'logrocket'
+
+import {ANONYMOUS_USERNAME, PROFILE_CONFIG_KEY} from '../constants'
+import {storeConfig} from '../config'
+
+import CustomSignIn from './custom-sign-in'
+import CustomHeaderBar from './custom-header-bar'
+
+export default function anonymousAuthenticator (Comp) {
+  return class extends Component {
+    constructor (props) {
+      super(props)
+
+      this.handleAnonymousLogin = this.handleAnonymousLogin.bind(this)
+      this.handleProfileChange = this.handleProfileChange.bind(this)
```

Comment:
I somewhat expected handleProfileChange to do the LoadProfile().  Any reason it doesn't?

Reply:
I use `loadProfile()` as passed as a prop from `Application` on line 20--did you expect to see it used in a different way or as a method of `anonymousAuthenticator` rather than passed as a prop?

Reply:
Conceptually I just expect handleProfileChange to "handle" loading the profile. It's fine as is, I was just curious.

Reply:
Ah, I see. That's a good point--I had made `handleProfileChange` to be a counterpart to Amplify's `withAuthenticator.changeUserProfile` which also doesn't use `loadProfile`, which makes sense within Amplify's structure but maybe not within this less intense `anonymousAuthenticator`. I'll explore that structure a bit more--thanks for pointing that out!